src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/kudu_client.py

```diff
@@ -113,9 +113,24 @@ def install_node_dependencies(self):
             'command': 'npm install',
             'dir': r'site\wwwroot'
         }
-        response = requests.post(self.__scm_url + '/api/command', data=json.dumps(payload),
-                                 headers=self.__get_application_json_headers())
-        HttpResponseValidator.check_response_status(response)
+        # npm install can take a very long time to complete. By default, Azure's load balancer will terminate
+        # connections after 230 seconds of no inbound or outbound packets. This timeout is not configurable.
+        try:
+            response = requests.post(self.__scm_url + '/api/command', data=json.dumps(payload),
+                                     headers=self.__get_application_json_headers())
+            HttpResponseValidator.check_response_status(response)
+        except CLIError as e:
+            if response.status_code == 500 and 'The request timed out.' in response.text:
+                self.__logger.warn('npm install is taking longer than expected and did not finish within the '
```

Comment:
Why are you not just using `logger.warn` like you do elsewhere? I've not seen this `self.__logger.warn` syntax used elsewhere in the CLI.

Reply:
Our module handles a wide array of tasks to get a bot up and running, so we refactored the code into multiple classes. Some commands perform a variety of calls to Kudu, instead of recreating the required info multiple times and passing it into static methods, we made a class that would store the necessary data (e.g. scm url, credentials). 

I added more logging and [`KuduClient.__logger`](https://github.com/Azure/azure-cli/pull/8035/files#diff-23f7ee5776637788fd8a28c5ab63cb3bR31) in the 0.1.3 PR (#8035) so that the public methods wouldn't need to take an additional argument for every call.

Since this code is not accessible by the user and we currently don't use any extensions with the latest version of botservice, I could change the structure to get it more inline with the overall structure of the CLI.

Reply:
Okay, no worries. Just curious.